web-ng/src/app/components/map/map.component.ts

```diff
@@ -163,42 +138,109 @@ export class MapComponent implements OnInit, AfterViewInit, OnDestroy {
     }
   }
 
-  onFeatureClick(featureId: string | null) {
-    this.navigationService.setFeatureId(featureId);
+  private deleteMarkers() {
+    for (const marker of this.markers) {
+      marker.setMap(null);
+    }
+    this.markers = [];
   }
 
-  isLocationFeature(feature: Feature) {
-    return feature instanceof LocationFeature;
+  private clearGoogleMapDataLayer() {
+    this.map.data.forEach(f => this.map.data.remove(f));
+  }
+
+  private addMarkersAndGeoJsonsToMap(
+    project: Project,
+    features: List<Feature>
+  ) {
+    features.forEach(feature => {
+      if (feature instanceof LocationFeature) {
+        this.addMarkerToMap(project, feature);
+      }
+      if (feature instanceof GeoJsonFeature) {
+        this.addGeoJsonToMap(feature);
+      }
+    });
   }
 
-  createMarkerOptions(
-    feature: LocationFeature,
-    focusedFeatureId: string,
-    project: Project
-  ): google.maps.MarkerOptions {
-    // Icon is not yet an input for <map-marker>, this is the only way to change icon for now.
-    // Consider break this down when more inputs are available for <map-marker>.
-    const normalScale = 30;
-    const enlargedScale = 50;
+  private addMarkerToMap(project: Project, feature: LocationFeature) {
     const color = project.layers.get(feature.layerId)?.color;
     const icon = {
       url: getPinImageSource(color),
       scaledSize: {
-        width: feature.id === focusedFeatureId ? enlargedScale : normalScale,
-        height: feature.id === focusedFeatureId ? enlargedScale : normalScale,
+        width: normalIconScale,
+        height: normalIconScale,
       },
     } as google.maps.Icon;
-
-    return {
+    const options: google.maps.MarkerOptions = {
+      map: this.map._googleMap,
       position: new google.maps.LatLng(
         feature.location.latitude,
         feature.location.longitude
       ),
       icon,
-    } as google.maps.MarkerOptions;
+      draggable: false,
+    };
+    const marker = new google.maps.Marker(options);
+    marker.addListener('click', () => {
+      this.focusMarker(marker);
+      this.navigationService.setFeatureId(feature.id);
+    });
+    marker.addListener('dragend', (event: google.maps.MouseEvent) => {
+      const newFeature = new LocationFeature(
+        feature.id,
+        feature.layerId,
+        new firestore.GeoPoint(event.latLng.lat(), event.latLng.lng())
+      );
+      this.featureService.updatePoint(newFeature);
+    });
+    this.markers.push(marker);
   }
 
-  private clearGoogleMapDataLayer() {
-    this.map.data.forEach(f => this.map.data.remove(f));
+  private focusMarker(marker: google.maps.Marker | null) {
```

Comment:
Iirc we've been using "selected" and "deselected", can we use the same terms here for consistency?

Also, consider splitting this into two methods, "selectMarker" and "deselectMarker".

Reply:
Renamed. If we split into "selectMarker" and "deselectMarker". Then we have to 
1. remember to deselect previous selected marker when selecting marker
2. make sure we deselect before select because select will assign selectedMarker, then we won't know who was previously selected.
So these two are basically still coupled. So I'll keep it in one method.